lib/reek/smells/smell_warning.rb

```diff
@@ -17,6 +17,8 @@ class SmellWarning
       attr_reader :context, :lines, :message, :parameters, :smell_detector, :source
       def_delegators :smell_detector, :smell_category, :smell_type
 
+      DETAIL_KEYS = %i(message lines context source parameters)
```

Comment:
Actually just see my comment directly below regarding "matches_smell_details?", that should make everthing clear ;)